optimum/neuron/distributed/base.py

```diff
@@ -525,7 +529,48 @@ def load_model_checkpoint(cls, model: "PreTrainedModel", load_dir: Union[str, Pa
 
         if (load_dir / TENSOR_PARALLEL_SHARDS_DIR_NAME).is_dir():
             cls.load_model_sharded_checkpoint(model, load_dir)
-        elif (load_dir / WEIGHTS_NAME).is_file():
-            cls.load_model_regular_checkpoint(model, load_dir)
         else:
-            raise FileNotFoundError(f"Could not find a checkpoint file under {load_dir.as_posix()}.")
+            raise FileNotFoundError(f"Could not find a sharded checkpoint directory under {load_dir.as_posix()}.")
+
+    @classmethod
+    def load_optimizer_sharded_checkpoint(cls, optimizer: "torch.optim.Optimizer", load_dir: Union[str, Path]):
+        from neuronx_distributed.optimizer import NeuronZero1Optimizer
```

Comment:
Any particular reason why the import happens here ? The method is not guarded by `@requires_neuronx_distributed`, and you have a check at the top of the file where you already import `parallel_layers`

Reply:
There are two ways of doing this;

1. Importing at the top of the file after having checked that the package was available
2. Do it on a per-function basis with the requires decorator

I was doing kind of both in the same file which is not great you're right, fixed it so follow the second way!